packages/DataGrid/src/DataGrid.js

```diff
@@ -0,0 +1,628 @@
+      refEnd.current &&
+      refScrollGrid.current.scrollTop + refScrollGrid.current.offsetHeight >= refScrollGrid.current.scrollHeight
+    ) {
+      refEnd.current.onScrollBarReachedBottom(true);
+    } else if (refEnd.current) {
+      refEnd.current.onScrollBarReachedBottom(false);
+    }
+  }, []);
+
+  const handleScrollStickyColumns = React.useCallback(
+    parameters => {
+      const { scrollTop /* scrollUpdateWasRequested */ } = parameters;
+
+      // prevent rescrolling when this scrollbar gets sync with the one in the main grid
+      if (refScrollHappenedBy.current === null) {
+        refScrollHappenedBy.current = "handleScrollStickyColumns";
+        if (refScrollGrid.current) {
+          refScrollGrid.current.scrollTo({ left: 0, top: scrollTop });
+        }
+      } else {
+        refScrollHappenedBy.current = null;
+      }
+    },
+    [refScrollGrid]
+  );
+
+  const handleRefCell = React.useCallback(({ columnIndex, rowIndex }) => {
+    return React.useCallback(
+      ref => {
+        const key = `${columnIndex}${rowIndex}`;
+        refsCell.current.keys[key] = ref;
+        refsCell.current.rows[rowIndex] = Array.isArray(refsCell.current.rows[rowIndex])
+          ? refsCell.current.rows[rowIndex].concat(key)
+          : [key];
+      },
+      [columnIndex, rowIndex]
+    );
+  }, []);
+
+  React.useEffect(() => {
+    if (!refContainer.current) return;
+
+    refScrollHeader.current = refContainer.current.querySelector(`.${gridId}-header`);
+    refScrollStickyColumns.current = refContainer.current.querySelector(`.${gridId}-sticky-columns`);
+    refScrollGrid.current = refContainer.current.querySelector(`.grid-${gridId}`);
+  }, [gridId]);
+
+  React.useEffect(() => {
+    const scrollContainer =
+      refContainer.current && refContainer.current.querySelector(`.grid-${gridId} [role="row"]`).parentElement;
+
+    if (!scrollContainer) return;
+    // https://stackoverflow.com/questions/21064101/understanding-offsetwidth-clientwidth-scrollwidth-and-height-respectively
+    // https://davidwalsh.name/detect-scrollbar-width
+    setScrollBarWidth(() => scrollContainer.offsetWidth - scrollContainer.clientWidth);
+  }, [gridId, isIdle]);
+
+  function handleFocusGrid() {
+    const $isBlur = refContainer.current.querySelector(".grid--is-blur");
+    if ($isBlur) $isBlur.classList.remove("grid--is-blur");
+
+    if (gridShouldHaveFocus) {
+      setGridShouldHaveFocus(() => {
+        return false;
+      });
+    }
+  }
+
+  React.useEffect(() => {
+    if (data.length && pageSize === null) {
+      setPageSize(() => data.length);
+    }
+  }, [data.length, pageSize]);
+
+  React.useEffect(() => {
+    refPrevLastScrollHeight.current = refScrollGrid.current.scrollHeight;
+  }, []);
+
+  React.useEffect(() => {
+    // this is required to readjust the active highlight
+    // after any rerender
+    if (refPrevLastScrollHeight.current && refPrevLastScrollHeight.current < refScrollGrid.current.scrollHeight) {
```

Comment:
When adding new rows to the table like you would do when clicking LoadMore, the onScroll event will not get notified, so to trick it I'm just tracking the previous height and if I notices has changed I move the scrollbar by 1px which triggers the onScroll event.